File: Swift/1/100_Doors.swift
import Foundation


enum doorStatus : String {
    case Opened = "Closed!"
    case Closed = "Open!"
}

var doorList = [doorStatus](repeating: doorStatus.Closed, count: 100)



for doorCount in 1...100 {
    let strideQuantity = stride(from: doorCount - 1, to: 100, by: doorCount)
    
     strideQuantity.map() {
        doorList[$0] = doorList[$0] == .Opened ? .Closed : .Opened
    }
}

for (indexCount, itemCount) in doorList.enumerated() {
    print("Door \(indexCount+1) is \(itemCount.rawValue)")
}

File: C/1/100_Doors.c
#include <stdio.h>

#define doorCount 100

int main(int argc, const char * argv[]) {
    int openDoors[doorCount] = {0};
    int *doorPointer, *doorLimit = openDoors + doorCount;
    int doorPass;
    
    for (doorPass=doorCount; (doorPass); doorPass--) {
        for (doorPointer = openDoors + (doorPass - 1); (doorPointer < doorLimit); doorPointer += doorPass) {
            (*doorPointer) ^= 1;
        }
    }
    
    for (doorPointer = openDoors; (doorPointer != doorLimit); doorPointer++) {
        printf("Door %ld is %s\n", (doorPointer - openDoors) + 1, (*doorPointer) ? "Opened!" : "Closed!");
    }
}
